test/framework/clusterctl/client.go

```diff
@@ -0,0 +1,155 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterctl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+	clusterctllog "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl/logger"
+)
+
+// Provide E2E friendly wrappers for the clusterctl client library.
+
+// InitInput is the input for Init.
+type InitInput struct {
+	LogPath                 string
+	ClusterctlConfigPath    string
+	KubeconfigPath          string
+	CoreProvider            string
+	BootstrapProviders      []string
+	ControlPlaneProviders   []string
+	InfrastructureProviders []string
+}
+
+// Init calls clusterctl init with the list of providers defined in the local repository
+func Init(ctx context.Context, input InitInput) {
+	By(fmt.Sprintf("clusterctl init --core %s --bootstrap %s --control-plane %s --infrastructure %s",
+		input.CoreProvider,
+		strings.Join(input.BootstrapProviders, ", "),
+		strings.Join(input.ControlPlaneProviders, ", "),
+		strings.Join(input.InfrastructureProviders, ", "),
+	))
+
+	initOpt := clusterctlclient.InitOptions{
+		Kubeconfig:              input.KubeconfigPath,
+		CoreProvider:            input.CoreProvider,
+		BootstrapProviders:      input.BootstrapProviders,
+		ControlPlaneProviders:   input.ControlPlaneProviders,
+		InfrastructureProviders: input.InfrastructureProviders,
```

Comment:
nit. InfrastructureProviders can be a string instead of a list as in ConfigClusterInput.

Reply:
clusterctl fully support multitenancy, so the library must use lists because in case of multitenancy there can be more that one infrastructure provider in a Cluster.